packages/extension/src/ui/features/accountNfts/alephium-nft.service.ts

```diff
@@ -103,11 +103,14 @@ async function getNftMetadataz(nftIds: string[], explorerProvider: ExplorerProvi
     }
   }
 
-  const newNftMetadataz = (await Promise.all(
-    chunk(nftIdsWithoutMetadata, maxSizeTokens).map((ids) =>
-      explorerProvider.tokens.postTokensNftMetadata(ids)
-    )
-  )).flat()
+  let newNftMetadataz: NFTMetadata[] = []
+  if (nftIdsWithoutMetadata.length !== 0) {
+    newNftMetadataz = (await Promise.all(
+      chunk(nftIdsWithoutMetadata, maxSizeTokens).map((ids) =>
+        explorerProvider.tokens.postTokensNftMetadata(ids)
+      )
+    )).flat()
+  }
 
   for (const newNftMetadata of newNftMetadataz) {
     await storeImmutable(`${newNftMetadata.id}-nft-metadata`, newNftMetadata)
```

Comment:
1. Will it be a problem that we create too many keys in the local storage?
2. Maybe it's better to use prefixes than postfixes?

Reply:
> Will it be a problem that we create too many keys in the local storage?

There isn't a limit to the number of keys, but local storage dose have a size constrains (10MB) which will put an upper limit to that. For token metadata, each entry is around 0.2~0.3kb, and we would need around a low tens of thousands of tokens to exceed the size limit (together with other things such as tx history). 
From what I read the individual key access is fast.

> Maybe it's better to use prefixes than postfixes?

I don't have preference here. If there are good benefits with postfixes I will update. (seems that it won't affect performance)

Reply:
Prefix is more readable and easier to sort keys based on key types.

Reply:
I will update